distributed/client.py

```diff
@@ -3961,6 +3961,39 @@ async def _get_task_stream(
         else:
             return msgs
 
+    async def _register_scheduler_plugin(self, plugin=None, idempotent=False, **kwargs):
+        if isinstance(plugin, type):
+            plugin = plugin(**kwargs)
+
+        return await self.scheduler.register_scheduler_plugin(
+            plugin=dumps(plugin, protocol=4)
+        )
+
+    def register_scheduler_plugin(self, plugin=None, idempotent=False, **kwargs):
+        """Register a scheduler plugin.
+
+        See https://distributed.readthedocs.io/en/latest/plugins.html
+
+        Parameters
+        ----------
+        plugin : SchedulerPlugin
+            Plugin class (not instance) to pass to the scheduler.
```

Comment:
It looks like `plugin` can be either a class or instance (which is the behavior we want)